### PR TITLE
Update dashboard

### LINKED ISSUE #270: DCD frontend: upate dashboard according to latest sketch
the requested changes are:

- center search field
- add recently used projects, using cookies for now

### PR DESCRIPTION
**Pull request by valentin-krasontovitsch**
Closes #270

### CODE REVIEW

File: frontend/src/Views/DashboardView.tsx [hunk 1]
```diff
@@ -23,19 +30,38 @@ const ProjectDropdown = styled(SingleSelect)`
...
 const DashboardView = () => {
     const navigate = useNavigate()
 
     const [projects, setProjects] = useState<any[]>()
+    const [recentProjects, setRecentProjects] = useState<Components.Schemas.ProjectDto[] | any>()
+
+    const getRecentProjects =
+    (projects: Components.Schemas.ProjectDto[]) => {
+
+        const recentProjectsWithTimeStamp = projects.map( project => {
```
Comment (aronhoyer) [inline frontend/src/Views/DashboardView.tsx:49]: let's have a look at how we're storing recently accessed projects.

**Review by aronhoyer — CHANGES_REQUESTED** [review]

File: frontend/src/Components/RecentProjects.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+
+interface Props {
+    projects: Components.Schemas.ProjectDto[]
+}
+
+const RecentProjects = ({ projects }: Props) => {
+
+    const renderProjectDG = (phase: Components.Schemas.ProjectPhase | undefined) => {
+        const backgroundColor = phase === undefined ?
```
[the remark this answers is not in the document]
Comment (valentin-krasontovitsch) [reply frontend/src/Components/RecentProjects.tsx:65]: thank you, fixed! love the fuzzyness of falsy : )

File: frontend/src/Components/RecentProjects.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+`
+
+
+interface Props {
+    projects: Components.Schemas.ProjectDto[]
+}
+
+const RecentProjects = ({ projects }: Props) => {
+
+    const renderProjectDG = (phase: Components.Schemas.ProjectPhase | undefined) => {
```
[the remark this answers is not in the document]
Comment (valentin-krasontovitsch) [reply frontend/src/Components/RecentProjects.tsx:64]: ah, nice, thank you, fixed!

File: frontend/src/Components/RecentProjects.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        const backgroundColor = phase === undefined ?
+            tokens.colors.ui.background__danger.rgba :
+                tokens.colors.ui.background__info.rgba
+        const labelText = phase === undefined ? 'TBD' : ProjectPhaseNumberToText(phase)
+        return <ProjectDG style={{background: backgroundColor}} variant="caption">
+            {labelText}
+        </ProjectDG>
+    }
+
+    const renderDescription = (description: string | null | undefined) => {
```
[the remark this answers is not in the document]
Comment (valentin-krasontovitsch) [reply frontend/src/Components/RecentProjects.tsx:74]: fixed

File: frontend/src/Components/RecentProjects.tsx [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+            return null
+        }
+        return (
+            <Typography variant="caption" lines={4}>
+                {description}
+            </Typography>
+        )
+    }
+
+    if (!projects || projects.length === 0) return null
```
[the remark this answers is not in the document]
Comment (valentin-krasontovitsch) [reply frontend/src/Components/RecentProjects.tsx:85]: thanks again! such a yummy feature ^^ fixed.

File: frontend/src/Components/RecentProjects.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (valentin-krasontovitsch) [reply frontend/src/Components/RecentProjects.tsx:74]: ah i had to keep the null in there, as `description` has the type `string? | null` in `types.d.ts` (and i get an error if i remove it)

File: frontend/src/Components/RecentProjects.tsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+    return (
+        <Wrapper>
+            <RecentProjectTitle variant="h3">
+                Recently used Projects
+            </RecentProjectTitle>
+            <RecentProjectCardWrapper>
+                {projects.map((project) => { return (
```
[the remark this answers is not in the document]
Comment (valentin-krasontovitsch) [reply frontend/src/Components/RecentProjects.tsx:93]: thanks, fixed! those details around arrow functions and when you can skip the return, i'm always hazy on them...

[the remark this answers is not in the document]
Comment (valentin-krasontovitsch) [reply frontend/src/Components/RecentProjects.tsx:93]: thanks, fixed! those details around arrow functions and when you can skip the return, i'm always hazy on them...

File: frontend/src/Views/DashboardView.tsx (hunk 1 above)
Comment (valentin-krasontovitsch) [reply frontend/src/Views/DashboardView.tsx:49]: sure thing!

**Review by aronhoyer — DISMISSED** [review]
looks good! we'll update storage strategy once the demo is done.

merge once we resolve snyk error.

**Review by aronhoyer — APPROVED** [review]

**Pull request merged**